### DISCUSSION TITLE
Failed to read Font Size when the content is nested with other styling.

### DISCUSSION THREAD

**Discussion by Arik14** [ueberdosis/tiptap#1815] [General Questions] [answered]
Hello,

I have another problem here.
This is the content I put into the instance of Tiptap Editor. This is before the Editor open.

![Screenshot from 2021-08-30 08-58-30](https://user-images.githubusercontent.com/54788602/131275359-ecaeab00-6c48-4d70-8724-82580ece8969.png)
And this one is when the Editor is open. It seems like parseHTML fail to recognize value of font size, when the span has nested child with other styling (font-family in this case). font-size null is the default in my custom extension of Font Size.

![Screenshot from 2021-08-30 08-58-51](https://user-images.githubusercontent.com/54788602/131275437-376ad88d-88d5-40e5-80a5-26e2c5f9bbc8.png)
But when I flip the nested order of those spans, it worked and the font size value is correct.

![Screenshot from 2021-08-30 08-56-53](https://user-images.githubusercontent.com/54788602/131275508-adb7d719-3e96-48b6-a9fd-0bbf6d56ad89.png)

In previous version of Tiptap, it worked with nested span. I have text with similar structure html span of span with various styling generate by previous Tiptap. But when I apply new Tiptap, it seems like my custom extension fail to recognize font size if it nested and it's child of another span with different styling.
If you want to play with my code, here is my sandbox https://codesandbox.io/s/wandering-wood-etvm7?file=/src/components/HelloWorld.vue. You can click button Toggle Editor to see what happen.

Please kindly help me.
Thank you.

**Comment by WangHaohui-org** [comment]
plz , help me !

**Reply by WangHaohui-org** [reply]
I have the same problem as you

**Reply by CarlMenke** [reply]
did you ever find a solution?

**Reply by nperez0111** [reply, marked as answer]
nested spans do not work because prosemirror (our underlying library) strips it by default. The correct way to specify this would be with a single `span` tag like:

`<span style="font-size: 40px; font-family: Arial">INNER TEXT</span>`

**Reply by ghlLarney** [reply]
Even if there's text content in between nested tags?
```html
<p>
    <span style="font-size: 14pt; font-family: arial, helvetica, sans-serif;">
        text1 <span style="color: #3366ff;">text2</span> text3<br />
        <br />
        text 4
    </span>
</p>
```

gets converted to this
```html
<p>
    <span style="font-size: 14pt; font-family: arial, helvetica, sans-serif;">text1 
  	</span>
  	<span style="color: #3366ff;">text2
  	</span> 
  	to your analytics account as a ‘view only’ user that will enable us to view without you having to share any logins! :)<br />
    <br />
	text4
</p>
```

**Reply by deki23** [reply]
Is there a way to somehow extend this through schema and override prosemirror behavior?

**Comment by bdbch** [comment]
Reopened this discussion as it seems this is still an issue for people. See the answer from @nperez0111 regarding Prosemirror's stripping behavior to understand what the current issue is.

I myself didn't look to deep into this so I can't 100% say where this is happening right now and if we can avoid it on Tiptap's side.

---

**Edit**: Actually I closed this and created an issue for this: https://github.com/ueberdosis/tiptap/issues/5720

Please discuss inside this issue instead of this discussion.
